apps/remix/app/components/forms/support-ticket-form.tsx

```diff
@@ -0,0 +1,144 @@
+import { useState } from 'react';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Trans, useLingui } from '@lingui/react/macro';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import { trpc } from '@documenso/trpc/react';
+import { Button } from '@documenso/ui/primitives/button';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from '@documenso/ui/primitives/form/form';
+import { Input } from '@documenso/ui/primitives/input';
+import { Textarea } from '@documenso/ui/primitives/textarea';
+import { useToast } from '@documenso/ui/primitives/use-toast';
+
+const ZSupportTicketSchema = z.object({
+  email: z.string().email('Please enter a valid email address'),
+  subject: z.string().min(3, 'Subject is required'),
+  message: z.string().min(10, 'Message must be at least 10 characters'),
+});
+
+type TSupportTicket = z.infer<typeof ZSupportTicketSchema>;
+
+export type SupportTicketFormProps = {
+  email: string;
+  onSuccess?: () => void;
+  onClose?: () => void;
+};
+
+export const SupportTicketForm = ({ email, onSuccess, onClose }: SupportTicketFormProps) => {
+  const { t } = useLingui();
+  const { toast } = useToast();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { mutateAsync: submitSupportTicket } = trpc.profile.submitSupportTicket.useMutation();
+
+  const form = useForm<TSupportTicket>({
+    resolver: zodResolver(ZSupportTicketSchema),
+    defaultValues: {
+      email: email || '',
+      subject: '',
+      message: '',
+    },
+  });
+
+  const onSubmit = async (values: TSupportTicket) => {
+    setIsLoading(true);
```

Comment:
We don't need to handle a loading state since react-hook-form has one for us in form.formState.isSubmitting.